src/net.h

```diff
@@ -781,6 +781,8 @@ class CConnman
     };
 
     void Init(const Options& connOptions) {
```

Comment:
Missing `EXCLUSIVE_LOCKS_REQUIRED(!m_total_bytes_sent_mutex)`

Reply:
Agree. I've verified this PR rebased on top of the current master (f60a63cc5f16b738d9d2ada3f10b27cf999df323):
```
$ ./configure CC=clang CXX=clang++ CXXFLAGS='-Wthread-safety -Wthread-safety-negative'
$ make clean
$ make 2>&1 | grep m_total_bytes_sent_mutex
```
ends with warnings
```
./net.h:781:13: warning: acquiring mutex 'm_total_bytes_sent_mutex' requires negative capability '!m_total_bytes_sent_mutex' [-Wthread-safety-negative]
            LOCK(m_total_bytes_sent_mutex);
```

To silent all of them, the following patch works:
```diff
--- a/src/net.h
+++ b/src/net.h
@@ -760,7 +760,8 @@ public:
         bool m_i2p_accept_incoming;
     };
 
-    void Init(const Options& connOptions) {
+    void Init(const Options& connOptions) EXCLUSIVE_LOCKS_REQUIRED(!m_total_bytes_sent_mutex)
+    {
         AssertLockNotHeld(m_total_bytes_sent_mutex);
 
         nLocalServices = connOptions.nLocalServices;
@@ -791,7 +792,7 @@ public:
 
     CConnman(uint64_t seed0, uint64_t seed1, AddrMan& addrman, bool network_active = true);
     ~CConnman();
-    bool Start(CScheduler& scheduler, const Options& options);
+    bool Start(CScheduler& scheduler, const Options& options) EXCLUSIVE_LOCKS_REQUIRED(!m_total_bytes_sent_mutex);
 
     void StopThreads();
     void StopNodes();
```

Reply:
Reproduced the issue and confirm that this patch works.

Reply:
Done. Thanks for the reviews and suggestions.